django_project/certification/views/certificate.py

```diff
@@ -730,3 +730,93 @@ def regenerate_all_certificate(request, **kwargs):
+    """View to issue all Certificates."""
+
+    context_object_name = 'certificate'
+    template_name = 'certificate/create.html'
+
+    def get_success_url(self):
+        """Define the redirect URL.
+
+        After successful creation of the object, the User will be redirected
+        to the Course detail page.
+
+       :returns: URL
+       :rtype: HttpResponse
+       """
+
+        return reverse('course-detail', kwargs={
+            'project_slug': self.project_slug,
+            'organisation_slug': self.organisation_slug,
+            'slug': self.course_slug
+        })
+
+    def get_context_data(self, **kwargs):
+        """Get the context data which is passed to a template.
+
+        :param kwargs: Any arguments to pass to the superclass.
+        :type kwargs: dict
+
+        :returns: Context data which will be passed to the template.
+        :rtype: dict
+        """
+
+        context = super(
+            IssueAllCerts, self).get_context_data(**kwargs)
+        context['course'] = Course.objects.get(slug=self.course_slug)
+        context['attendee'] = Attendee.objects.get(pk=self.pk)
+        return context
+
+    def get_form_kwargs(self):
+        """Get keyword arguments from form.
+
+        :returns keyword argument from the form
+        :rtype: dict
+        """
+
+        kwargs = super(IssueAllCerts, self).get_form_kwargs()
+        self.project_slug = self.kwargs.get('project_slug', None)
+        self.organisation_slug = self.kwargs.get('organisation_slug', None)
+        self.course_slug = self.kwargs.get('course_slug', None)
+        self.pk = self.kwargs.get('pk', None)
+        self.course = Course.objects.get(slug=self.course_slug)
+        self.attendee = Attendee.objects.get(pk=self.pk)
+        kwargs.update({
+            'user': self.request.user,
+            'course': self.course,
+            'attendee': self.attendee,
+        })
+        return kwargs
+
+    def form_valid(self, form):
+        """Save new created certificate
+
+        :param form
+        :type form
+
+        :returns HttpResponseRedirect object to success_url
+        :rtype: HttpResponseRedirect
+
+        We check that there is no referential integrity error when saving."""
+
+        try:
+            super(IssueAllCerts, self).form_valid(form)
+
+            # Update organisation credits every time a certificate is issued.
+            organisation = \
+                CertifyingOrganisation.objects.get(
+                    slug=self.organisation_slug)
+            remaining_credits = \
+                organisation.organisation_credits - \
+                organisation.project.certificate_credit
+            organisation.organisation_credits = remaining_credits
+            organisation.save()
+
+            return HttpResponseRedirect(self.get_success_url())
+        except IntegrityError:
+            return ValidationError(
+                'ERROR: Certificate already exists!')
```

Comment:
__[Flake8]__

[W292] no newline at end of file


<!-- Comment by [SideCI](https://sideci.com) -->